suave/builder/session_manager.go

```diff
@@ -37,6 +39,38 @@ type Config struct {
 	MaxConcurrentSessions int
 }
 
+func getEnvAsUint64(key string, defaultVal uint64) uint64 {
+	valStr := os.Getenv(key)
+	if val, err := strconv.ParseUint(valStr, 10, 64); err == nil {
+		return val
+	}
+	return defaultVal
+}
+
+func getEnvAsInt(key string, defaultVal int) int {
+	valStr := os.Getenv(key)
+	if val, err := strconv.Atoi(valStr); err == nil {
+		return val
+	}
+	return defaultVal
+}
+
+func getEnvAsDuration(key string, defaultVal time.Duration) time.Duration {
+	valStr := os.Getenv(key)
+	if val, err := time.ParseDuration(valStr); err == nil {
+		return val
+	}
+	return defaultVal
+}
+
+func NewConfigFromEnv() *Config {
```

Comment:
I would prefer to use a pattern like this:
```
func NewConfig() *Config {
   cfg := &Config{ // default config
        GasCeil: 1000000000000000000,
        SessionIdleTimeout: 5*time.Second,
   }

   // apply envs if exit
   var err error
   if valStr := os.GetEnv("GAS_CEIL"); valStr != "" {
       if cfg.GasCeil, err = strconv.Atoi(valStr); err != nil {
           // handle error
       }
   }
}
```
I like to make the default explicit